auth-starter/src/main/java/com/vaadin/auth/starter/VaadinAuthSecurityConfiguration.java

```diff
@@ -199,4 +237,30 @@ CompositeLogoutHandler getLogoutHandler() {
             return logoutHandler;
         }
     }
+
+    static class VaadinExpiredSessionStrategy
+            implements SessionInformationExpiredStrategy {
+
+        private static final Logger LOGGER = LoggerFactory
+                .getLogger(VaadinExpiredSessionStrategy.class);
+
+        @Override
+        public void onExpiredSessionDetected(
+                SessionInformationExpiredEvent event)
+                throws IOException, ServletException {
+            final var request = event.getRequest();
+            final var response = event.getResponse();
+            final var vaadinRequestParameter = request.getParameter("v-r");
+            final var redirectRoute = '/' + request.getContextPath();
+            if ("uidl".equals(vaadinRequestParameter)) {
+                LOGGER.debug("Session expired during UIDL request: setting "
+                        + "Vaadin-Refresh token");
+                response.getWriter().write("Vaadin-Refresh: " + redirectRoute);
```

Comment:
I'm wondering if 302 is the correct Status Code, or 403 has to be used like e.g. in HeartbeatHandler#L99 in flow to "kill" the push connection on the client.. that worked nicely for our applications with static landing pages, but I'm not sure of that would be the goal for redirect routes that are Vaadin views.

Reply:
That is the code used by the default Spring's expired-session strategy when a redirect URL is specified, so we're being consistent with it. Consider that the redirect happens just for non-UIDL requests, otherwise the `Vaadin-Regresh` token comes in action and Flow takes care of that.